src/sentry/models/activity.py

```diff
@@ -36,7 +36,7 @@ def get_activities_for_group(self, group: Group, num: int) -> Sequence[Group]:
         activity_qs = self.filter(group=group).order_by("-datetime")
 
         if not features.has("projects:issue-priority", group.project):
-            activity_qs.exclude(type=ActivityType.SET_PRIORITY.value)
+            activity_qs = activity_qs.exclude(type=ActivityType.SET_PRIORITY.value)
 
```

Comment:
In terms of filtering duplications, should we also consider `item.data` along with `item.type`, `item.ident`, and `item.user_id`?

Reply:
I think you're right, it should compare those values as well

Reply:
I have no idea why the duplication filtering is actually necessary though 🤔